react/tic-tac-toe/src/app/screens/Game/index.js

```diff
@@ -1,20 +1,65 @@
-import React from 'react';
+import React, { Component } from 'react';
+
+import { calculateWinner } from '../../../utils';
 
 import styles from './styles.module.scss';
 import Board from './components/Board';
+import Moves from './components/Moves';
 
-function Game () {
-  return (
-    <div className={styles.game}>
-      <div className={styles.gameBoard}>
-        <Board />
-      </div>
-      <div className={styles.gameInfo}>
-        <div>{/* status */}</div>
-        <ol>{/* TODO */}</ol>
+class Game extends Component {
+  state = {
+    history: [{ squares: Array(9).fill(null) }],
+    xIsNext: true,
+    stepNumber: 0
+  };
+
+  handleClick = i => {
+    const { history, xIsNext, stepNumber } = this.state;
+    const newHistoryPoint = history.slice(0, stepNumber + 1);
+    const current = newHistoryPoint[newHistoryPoint.length - 1];
+    const squares = current.squares.slice();
+    if (calculateWinner(squares) || squares[i]) {
+      return;
+    }
+
+    squares[i] = xIsNext ? 'X' : 'O';
+
+    this.setState(prevState => ({
+      history: newHistoryPoint.concat([{ squares }]),
+      stepNumber: newHistoryPoint.length,
+      xIsNext: !prevState.xIsNext
+    }));
+  };
+
+  handleJumpTo = step => {
+    this.setState({
+      stepNumber: step,
+      xIsNext: step % 2 === 0
+    });
+  };
+
+  handleViewMoves = history =>
+    history.map((_step, move) => <Moves key={`move-${move + 1}`} onClick={this.handleJumpTo} move={move} />);
+
+  render() {
+    const { history, xIsNext, stepNumber } = this.state;
+    const current = history[stepNumber];
+    const winner = calculateWinner(current.squares);
+
+    const status = winner ? `Winner: ${winner}` : `Next player: ${xIsNext ? 'X' : 'O'}`;
+
+    return (
+      <div className={styles.game}>
+        <div className={styles.gameBoard}>
+          <Board squares={current.squares} onClick={this.handleClick} />
+        </div>
+        <div className={styles.gameInfo}>
+          <div>{status}</div>
+          <ol>{this.handleViewMoves(history)}</ol>
```

Comment:
Move the mapping here instead, would be clearer to understand what's going on and might actually not make a huge difference in performance.